packages/core/src/modules/proofs/protocol/v1/V1ProofService.ts

```diff
@@ -996,6 +997,38 @@ export class V1ProofService extends ProofService<[IndyProofFormat]> {
     })
   }
 
+  public async getFormatData(
+    agentContext: AgentContext,
+    proofRecordId: string
+  ): Promise<GetFormatDataReturn<ProofFormat[]>> {
+    const [proposalMessage, requestMessage, presentationMessage] = await Promise.all([
+      this.findProposalMessage(agentContext, proofRecordId),
+      this.findRequestMessage(agentContext, proofRecordId),
+      this.findPresentationMessage(agentContext, proofRecordId),
+    ])
+
+    const indyProposeProof = proposalMessage?.indyProofProposal ?? undefined
+    const indyRequestProof = requestMessage?.indyProofRequest ?? undefined
+    const indyPresentProof = presentationMessage?.indyProof ?? undefined
```

Comment:
For the credentials API we actually return the raw JSON attachment data, without transforming it to a class instance. These methods do the JSON transform on it